File: src/main/java/catchla/yep/view/iface/IExtendedView.java
/*
 * Copyright (c) 2015. Catch Inc,
 */

package catchla.yep.view.iface;

import android.graphics.Rect;
import android.view.MotionEvent;
import android.view.View;

public interface IExtendedView {

    void setOnFitSystemWindowsListener(final OnFitSystemWindowsListener listener);

    void setOnSizeChangedListener(final OnSizeChangedListener listener);

    void setTouchInterceptor(final TouchInterceptor listener);

    interface OnFitSystemWindowsListener {
        void onFitSystemWindows(Rect insets);
    }

    interface OnSizeChangedListener {
        void onSizeChanged(View view, int w, int h, int oldw, int oldh);
    }

    interface TouchInterceptor {

        boolean dispatchTouchEvent(View view, MotionEvent event);

        boolean onInterceptTouchEvent(View view, MotionEvent event);

        boolean onTouchEvent(View view, MotionEvent event);

    }
}

File: src/main/java/catchla/yep/view/iface/IExtendedView.kt
/*
 * Copyright (c) 2015. Catch Inc,
 */

package catchla.yep.view.iface

import android.graphics.Rect
import android.view.MotionEvent
import android.view.View

interface IExtendedView {

    var touchInterceptor: TouchInterceptor?

    var onSizeChangedListener: OnSizeChangedListener?

    var onFitSystemWindowsListener: OnFitSystemWindowsListener?

    interface OnFitSystemWindowsListener {
        fun onFitSystemWindows(insets: Rect)
    }

    interface OnSizeChangedListener {
        fun onSizeChanged(view: View, w: Int, h: Int, oldw: Int, oldh: Int)
    }

    interface TouchInterceptor {

        fun dispatchTouchEvent(view: View, event: MotionEvent): Boolean

        fun onInterceptTouchEvent(view: View, event: MotionEvent): Boolean

        fun onTouchEvent(view: View, event: MotionEvent): Boolean

    }
}
